libraries/node-core-library/src/FileError.ts

```diff
@@ -0,0 +1,166 @@
+    this.line = options.line;
+    this.column = options.column;
+
+    // Manually set the prototype, as we can no longer extend built-in classes like Error, Array, Map, etc.
+    // https://github.com/microsoft/TypeScript-wiki/blob/main/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work
+    //
+    // Note: the prototype must also be set on any classes which extend this one
+    (this as any).__proto__ = FileError.prototype; // eslint-disable-line @typescript-eslint/no-explicit-any
+  }
+
+  /**
+   * Get the Unix-formatted the error message.
+   *
+   * @override
+   */
+  public toString(): string {
+    // Default to formatting in 'Unix' format, for consistency.
+    return this.getFormattedErrorMessage();
+  }
+
+  /**
+   * Get the formatted error message.
+   *
+   * @param options - Options for the error message format.
+   */
+  public getFormattedErrorMessage(options?: IFileErrorFormattingOptions): string {
+    return Path.formatFileLocation({
+      format: options?.format || 'Unix',
+      baseFolder: this._evaluateBaseFolder(),
+      pathToFormat: this.absolutePath,
+      message: this.message,
+      line: this.line,
+      column: this.column
+    });
+  }
+
+  private _evaluateBaseFolder(): string | undefined {
+    // If the environment variable isn't defined, we will return the project folder by default
+    if (!process.env.RUSHSTACK_FILE_ERROR_BASE_FOLDER) {
+      return this.projectFolder;
+    }
+
+    // Strip leading and trailing quotes, if present.
+    const rawBaseFolderEnvVariable: string = process.env.RUSHSTACK_FILE_ERROR_BASE_FOLDER.replace(
+      /^("|')|("|')$/g,
+      ''
+    );
+
+    const baseFolderTokenRegex: RegExp = /{([^}]+)}/g;
+    const result: RegExpExecArray | null = baseFolderTokenRegex.exec(rawBaseFolderEnvVariable);
+    if (!result) {
+      // No tokens, it's some absolute path. Return the value as-is.
+      return rawBaseFolderEnvVariable;
+    } else if (result.index !== 0) {
+      // Currently only support the token being first in the string.
+      throw new Error(
+        'The RUSHSTACK_FILE_ERROR_BASE_FOLDER environment variable contains text before ' +
+          `the token "${result[0]}".`
+      );
+    } else if (result[0].length !== rawBaseFolderEnvVariable.length) {
+      // Currently only support the token being the entire string.
+      throw new Error(
+        'The RUSHSTACK_FILE_ERROR_BASE_FOLDER environment variable contains text after ' +
+          `the token "${result[0]}".`
+      );
+    }
+
+    const token: string = result[1];
+    switch (token) {
+      case 'ABSOLUTE_PATH': {
+        return undefined;
+      }
+      case 'PROJECT_FOLDER': {
+        return this.projectFolder;
+      }
+      default: {
+        throw new Error(
+          `The RUSHSTACK_FILE_ERROR_BASE_FOLDER environment variable contains a token "${result[0]}", ` +
+            'which is not supported.'
+        );
+      }
+    }
```

Comment:
This could actually be cached too. Something like `static const getBaseFolderFunctionByBaseFolderEnvVar: Map<string | undefined, (fileError: FileError) => string | undefined>`